### DISCUSSION TITLE
MediaSizeLimitAttribute.cs Affected the use of upload

### DISCUSSION THREAD

**Discussion by DuskSleep** [OrchardCMS/OrchardCore#14232] [Q&A]
Trying to use this CMS
Found that cannot exceed 2MB on media
Even some 1MB cannot be uploaded
But trying to rewrite C# is still using the official version
Tried various web.config and appsettings.json and rewriting Program.cs

unable to resolve this issue

Solutions from other sites are too old

**Comment by dministro** [comment]
Hi @DuskSleep, Could you share your `appsettings.*.json` and version information here, and/or your source code in a public GH, BitBucket, etc. repository?

**Reply by DuskSleep** [reply]
appsettings.*.json 

{
  "DatabaseProvider": "Sqlite",
  "DocumentTable": "Document",
  "IdentityColumnSize": "Int64",
  "TableNameSeparator": "_",
  "FileSizeLimit": 4097152,
  "OrchardCore_Media": {
      // The accepted sizes for custom width and height.
      // When the 'UseTokenizedQueryString' is True (default) all sizes are valid.
      "SupportedSizes": [ 16, 32, 50, 100, 160, 240, 480, 600, 1024, 2048, 4096],

      // The number of days to store images in the browser cache.
      // NB: To control cache headers for module static assets, refer to the Orchard Core Modules Section.
      "MaxBrowserCacheDays": 30,

      // The number of days a cached resized media item will be valid for, before being rebuilt on request.
      "MaxCacheDays": 365,

      // The maximum size of an uploaded file in bytes. 
      // NB: You might still need to configure the limit in IIS (https://docs.microsoft.com/en-us/iis/configuration/system.webserver/security/requestfiltering/requestlimits/)
      "MaxFileSize": 90000000,
      
      // The path used when serving media assets.
      "AssetsRequestPath": "/media",

      // The path used to store media assets. The path can be relative to the tenant's App_Data folder, or absolute.
      "AssetsPath": "Media",

      // Whether to use a token in the query string to prevent disc filling.
      "UseTokenizedQueryString": true,

      // The list of allowed file extensions
      "AllowedFileExtensions": [

            // Images
            ".jpg",
            ".jpeg",
            ".png",
            ".gif",
            ".ico",
            ".svg",

            // Documents
            ".pdf", // Portable Document Format; Adobe Acrobat
            ".doc", // Microsoft Word Document
            ".docx",
            ".ppt", // Microsoft PowerPoint Presentation
            ".pptx",
            ".pps",
            ".ppsx",
            ".odt", // OpenDocument Text Document
            ".xls", // Microsoft Excel Document
            ".xlsx",
            ".psd", // Adobe Photoshop Document

            // Audio
            ".mp3",
            ".m4a",
            ".ogg",
            ".wav",

            // Video
            ".mp4", // MPEG-4
            ".m4v",
            ".mov", // QuickTime
            ".wmv", // Windows Media Video
            ".avi",
            ".mpg",
            ".ogv", // Ogg
            ".3gp", // 3GPP
        ],

      // The Content Security Policy to apply to assets served from the media library.
      "ContentSecurityPolicy" : "default-src 'self'; style-src 'unsafe-inline'",

      // The maximum chunk size when uploading files in bytes. If 0, no chunked upload is used. This is useful to work around request size limitations of a hosting environment.
      "MaxUploadChunkSize": 104857600,

      // The lifetime of temporary files created during upload. Defaults to 1 hour.
      "TemporaryFileLifetime": "01:00:00"
    },

Orchardcore version is 1.6.0

I'm not quite sure how to override the filesize limit on .net7

However, in my actual use, part of the image size is 1920*1080 and the JPG format is about 1MB, and it will also exceed the maximum size limit.

Part of the PNG format can be uploaded over 2.5MB.

After seeing the implementation, I don't understand why
